### DISCUSSION TITLE
Allow same locale under different domains

### DISCUSSION THREAD

**Discussion by andresgutgon** [vercel/next.js#21927] [Ideas]
**Describe the feature you'd like to request**

Hi, first of all thanks a lot for this project. Is awesome. I just started using and looks super nice. I'll try to explain my use case:
We need to build a page that is multi domain. NextJS support that but doesn't support having the same locale under 2 domains. For example:
```
example.fr
  |- fr (main locale no folder route)
  |-/en
example.be
  |-/fr
  |-/en
```

We have 2 use cases. 
1. First one is supporting countries where the same locale applies. Like for example Belgium and France where french is one of their national languages. 

2. The other use case is supporting the same language `/en` under different domains. The [assumption here](https://github.com/vercel/next.js/pull/18240) is that this is going to be duplicated content. But imagine you have a product that have different features based on the country you're. In that case you are explaining different things in the same language.

### Not sure duplicated content affects SEO 
According to this article [Google: Duplicate Content is Not a Negative Ranking Factor](https://www.searchenginejournal.com/google-duplicate-content-is-not-a-negative-ranking-factor/394321/) at the end of the article you can see [John Mueller (Webmaster Trends Analyst at Google)](https://www.linkedin.com/in/johnmu/?originalSubdomain=ch) talking about it. Please let me know if I'm understanding wrongly what he says. 

But in any case I think the use cases I exposed are valid concerns :)

**Describe the solution you'd like**

According to the [code here](https://github.com/vercel/next.js/pull/18274#discussion_r563285250) NextJs is raising an error if you try to configure `i18n` with multi domain and same locales. Not sure if others parts of NextJS need to be changed to allow this. As I said I'm not a NextJS expert :)

**Describe alternatives you've considered**

The alternative is to allow more than one domain with the same locale :smile:

**Comment by andresgutgon** [comment]
Any feedback here @Timer ?  :)

**Comment by elias-thok** [comment]
Has there been any updates on this?
I need this as well and I would like to solve it without using several deployments or a custom backend-server.

**Comment by pd8** [comment]
Also interested on this, any update on this?
Could we perhaps use something like... `fr-gb` for french language in uk or `en-fr` for english in french markets?

**Comment by debragail** [comment]
status?

**Comment by jaandrews** [comment]
I'm running into this as well. I'm working with a headless CMS that allows the user to attach new domains to content, so without a way to support multiple domains, it can't identify which [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) and [paths](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths) to use. It is possible to work around this by spinning up a new nextjs server for each domain with environmental variables to identify it, but this is less than ideal since they would all be using the same code and will lead to increased hosting costs in some cases.
